### PR TITLE
add optimal parameter configurations

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add tuned profile, udev script and cpu governor systemd service configurations.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5303
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: check-before-deployment.md [hunk 1]
```diff
@@ -259,126 +263,272 @@ Take the following steps to check whether THP is disabled. If not, follow the st
...
+
+    ```bash
+    cat /sys/kernel/mm/transparent_hugepage/enabled
+    ```
+
+    ```
+    always madvise [never]
+    ```
+
+7. Execute the following command to verify the I/O Scheduler of the disk where the data directory locates:
```
Comment (TomShawn) [inline check-before-deployment.md:506]:
```suggestion
7. Execute the following command to verify the I/O Scheduler of the disk where the data directory is located:
```

File: check-before-deployment.md [hunk 2]
```diff
@@ -259,126 +263,272 @@ Take the following steps to check whether THP is disabled. If not, follow the st
...
     ```
 
-2. Execute the `grubby` command to see the default kernel version:
-
     > **Note:**
     >
-    > Before you execute `grubby`, you should first install the `grubby` package.
+    > If `[always] madvise never` is output, THP is enabled. You need to disable it.
+
+2. Execute the following command to see the I/O Scheduler of the disk where the data directory locates. Assume that you create data directories on both sdb and sdc disks:
```
Comment (TomShawn) [inline check-before-deployment.md:270]:
```suggestion
2. Execute the following command to see the I/O Scheduler of the disk where the data directory is located. Assume that you create data directories on both sdb and sdc disks:
```

**Review by TomShawn — COMMENTED** [review]
Minor comments, Rest LGTM
@pepezzzz PTAL

**Comment by pepezzzz** [comment]
LGTM

**Review by TomShawn — APPROVED** [review]

**Pull request merged**